gnmi_server/server_test.go

```diff
@@ -2403,20 +2412,23 @@ func runTestSubscribe(t *testing.T, namespace string) {
 			c := client.New()
 			defer c.Close()
 			var gotNoti []client.Notification
+			var mu_gotNoti sync.Mutex
 			q.NotificationHandler = func(n client.Notification) error {
+				mu_gotNoti.Lock()
 				if nn, ok := n.(client.Update); ok {
 					nn.TS = time.Unix(0, 200)
 					gotNoti = append(gotNoti, nn)
 				} else {
 					gotNoti = append(gotNoti, n)
 				}
-
+				mu_gotNoti.Unlock()
 				return nil
 			}
-			go func() {
+			go func(t2 TestExec) {	
```

Comment:
Why is this variable passed in? Couldn't `tt` be used how it currently is?

Reply:
Because it reports data race between line2385 and this goroutine, by creating a local copy of tt using a closure, each goroutine now has its own copy of the variable, eliminating the data race.